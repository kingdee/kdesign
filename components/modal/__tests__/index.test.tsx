import React from 'react'
import { mount, render, ReactWrapper } from 'enzyme'
import Modal from '../index'
import { ModalType, ModalTypes } from '../modal'
// import { act } from 'react-dom/test-utils'
import mountTest from '../../../tests/shared/mountTest'
import { simulateEvent } from '../../../tests/shared/simulateEvent'
import ConfigProvider from '../../config-provider/index'
jest.useFakeTimers()
const setWrapperVisibleFC = () => {
  return (wrapper: ReactWrapper) => {
    wrapper.setProps({
      visible: true,
    })
    wrapper.update()
  }
}
const prefixClassName = 'kd-modal'
const classNameMap = {
  container: `.${prefixClassName}-container`,
  ok: `button.${prefixClassName}-ok-btn`,
  cancel: `button.${prefixClassName}-cancel-btn`,
  closeIcon: `.${prefixClassName}-close-icon`,
  mask: `.${prefixClassName}-mask`,
  containerBox: `.${prefixClassName}-container-box`,
  body: `.${prefixClassName}-body`,
  footer: `.${prefixClassName}-footer`,
}
const unmountHelp = (wrapper: ReactWrapper) => {
  expect(() => {
    wrapper.unmount()
  }).not.toThrow()
}
describe('Modal', () => {
  mountTest(Modal!)
  ModalTypes.forEach((type) => {
    mountTest(() => <Modal type={type} getContainer={false} />)
  })
  it('renders correctly', () => {
    expect(render(<Modal getContainer={false} />)).toMatchSnapshot()
    ModalTypes.forEach((type) => {
      expect(render(<Modal type={type} getContainer={false} />)).toMatchSnapshot()
    })
  })
  it('warns if type is wrong', () => {
    const mockWarn = jest.fn()
    jest.spyOn(console, 'warn').mockImplementation(mockWarn)
    const type = 'who am I' as any as ModalType
    render(<Modal type={type} getContainer={false} />)
    expect(mockWarn).toHaveBeenCalledTimes(1)
    expect(mockWarn.mock.calls[0][0]).toMatch("Warning: [kdesign]-modal: cannot found modal type 'who am I'")
  })

  it('should have displayName static property', () => {
    const wrapper = mount(<Modal></Modal>)
    expect((wrapper.type() as any).displayName).toBe('Modal')
  })
  const customedReactNodeProp = ['body', 'title', 'footer', 'titleIcon']
  const generatorCustomedReactNode = (prop: string) => {
    return [
      prop,
      () => {
        return <div className={`customed-${prop}`}></div>
      },
      `.customed-${prop}`,
    ]
  }
  it.each(customedReactNodeProp.map(generatorCustomedReactNode))(
    'should %s use right',
    (prop: string, CustomedNode, className) => {
      const wrapper = mount(<Modal {...{ [prop]: <CustomedNode /> }} />)
      expect(wrapper.exists(className)).toBe(true)
      expect(() => {
        wrapper.unmount()
      }).not.toThrow()
    },
  )
  it('should draggable use right', () => {
    ;[true, false].forEach((draggable) => {
      expect(render(<Modal draggable={draggable} getContainer={false} />)).toMatchSnapshot()
    })
  })
  it('should focusTriggerAfterClose use right', () => {
    const div = document.createElement('div')
    div.classList.add('first-test-div')
    const anotherDiv = document.createElement('div')
    anotherDiv.classList.add('second-test-div')
    const spyDivFocus = jest.spyOn(div, 'focus')
    const anotherSpyDivFocus = jest.spyOn(anotherDiv, 'focus')
    const activeSelectedElement = (div: HTMLElement) => {
      return Reflect.defineProperty(document, 'activeElement', {
        get() {
          return div
        },
        configurable: true,
      })
    }
    // kinds of close modal's type
    // click button
    activeSelectedElement(div)
    const onOk = () =>
      wrapper.setProps({
        visible: false,
      })
    const onCancel = () =>
      wrapper.setProps({
        visible: false,
      })
    const wrapper = mount(
      <Modal
        focusTriggerAfterClose
        visible={true}
        onOk={onOk}
        onCancel={onCancel}
        keyboard
        maskClosable
        closable
        getContainer={false}
      />,
    )
    const setWrapperVisible = setWrapperVisibleFC()
    wrapper.find(classNameMap.ok).simulate('click')
    wrapper.update()
    expect(spyDivFocus).toBeCalledTimes(1)
    activeSelectedElement(anotherDiv)
    setWrapperVisible(wrapper)
    wrapper.find(classNameMap.cancel).simulate('click')
    wrapper.update()
    expect(anotherSpyDivFocus).toBeCalledTimes(1)

    activeSelectedElement(div)
    setWrapperVisible(wrapper)
    wrapper.find(classNameMap.closeIcon).simulate('click')
    wrapper.update()
    expect(spyDivFocus).toBeCalledTimes(2)

    // keydown
    // esc
    activeSelectedElement(anotherDiv)
    setWrapperVisible(wrapper)
    simulateEvent(document.body, 'keydown', {
      key: 'Escape',
    })
    wrapper.update()
    expect(anotherSpyDivFocus).toBeCalledTimes(2)

    // enter
    activeSelectedElement(div)
    setWrapperVisible(wrapper)
    wrapper.find(classNameMap.containerBox).simulate('keydown', {
      key: 'Enter',
    })
    wrapper.update()
    expect(spyDivFocus).toBeCalledTimes(3)

    // click mask
    activeSelectedElement(anotherDiv)
    setWrapperVisible(wrapper)
    wrapper.find(classNameMap.mask).simulate('click')
    wrapper.update()
    expect(anotherSpyDivFocus).toBeCalledTimes(3)

    // IE Esc
    activeSelectedElement(div)
    setWrapperVisible(wrapper)
    simulateEvent(document.body, 'keydown', {
      key: 'Esc',
    })
    wrapper.update()
    expect(spyDivFocus).toBeCalledTimes(4)

    unmountHelp(wrapper)
  })

  it('should visible use right', () => {
    const onOk = () =>
      setTimeout(() => {
        wrapper.setProps({
          visible: false,
        })
      }, 5000)
    const onCancel = () =>
      setTimeout(() => {
        wrapper.setProps({
          visible: false,
        })
      }, 3000)
    const wrapper = mount(<Modal visible onOk={onOk} onCancel={onCancel} getContainer={false} />)
    expect(wrapper.exists(classNameMap.containerBox)).toBe(true)
    wrapper.find(classNameMap.ok).simulate('click')
    wrapper.update()
    jest.advanceTimersByTime(4000)
    expect(wrapper.exists(classNameMap.containerBox)).toBe(true)
    jest.advanceTimersByTime(1000)
    wrapper.update()
    console.log(wrapper.debug({ verbose: true }), wrapper.props())
    expect(wrapper.exists(classNameMap.container)).toBe(true)
    expect((wrapper.find(classNameMap.container).getDOMNode() as HTMLElement).style.display).toBe('')
    wrapper.setProps({
      visible: true,
    })
    wrapper.update()
    wrapper.find(classNameMap.cancel).simulate('click')
    wrapper.update()
    jest.advanceTimersByTime(2000)
    expect(wrapper.exists(classNameMap.containerBox)).toBe(true)
    jest.advanceTimersByTime(1000)
    wrapper.update()
    expect(wrapper.exists(classNameMap.containerBox)).toBe(true)

    unmountHelp(wrapper)
  })

  it('should keyboard use right', () => {
    const onOk = jest.fn(() =>
      wrapper.setProps({
        visible: false,
      }),
    )
    const onCancel = jest.fn(() =>
      wrapper.setProps({
        visible: false,
      }),
    )
    const wrapper = mount(<Modal visible onOk={onOk} onCancel={onCancel} keyboard />)

    simulateEvent(document.body, 'keydown', {
      key: 'Escape',
    })
    wrapper.update()
    expect(onOk).not.toHaveBeenCalled()
    expect(onCancel).toHaveBeenCalled()
    expect(wrapper.exists(classNameMap.container)).toBe(true)
    expect((wrapper.find(classNameMap.container).getDOMNode() as HTMLElement).style.display).toBe('')
    unmountHelp(wrapper)
  })

  it('should keydown enter can be used right', () => {
    const onOk = jest.fn(() =>
      wrapper.setProps({
        visible: false,
      }),
    )
    const onCancel = jest.fn(() =>
      wrapper.setProps({
        visible: false,
      }),
    )
    const wrapper = mount(<Modal visible onOk={onOk} onCancel={onCancel} keyboard />)

    wrapper.find(classNameMap.mask).simulate('keydown', {
      key: 'Enter',
    })
    wrapper.update()
    expect(onOk).not.toHaveBeenCalled()
    expect(onCancel).not.toHaveBeenCalled()

    wrapper.find(classNameMap.containerBox).simulate('keydown', {
      key: 'Enter',
    })
    expect(onOk).toHaveBeenCalled()
    expect(onCancel).not.toHaveBeenCalled()
    expect(wrapper.exists(classNameMap.container)).toBe(true)
    expect((wrapper.find(classNameMap.container).getDOMNode() as HTMLElement).style.display).toBe('')
    unmountHelp(wrapper)
  })

  it('should mask use right', () => {
    const wrapper = mount(<Modal mask={false} keyboard />)
    expect(wrapper.exists(classNameMap.mask)).toBe(false)

    wrapper.setProps({
      mask: true,
    })
    wrapper.update()
    expect(wrapper.exists(classNameMap.mask)).toBe(true)
    unmountHelp(wrapper)
  })

  it('should getContainer use right', () => {
    const mockWarn = jest.fn()
    jest.spyOn(console, 'warn').mockImplementation(mockWarn)
    const TestComp = () => {
      const ref = React.useRef(null)
      return (
        <div ref={ref}>
          <Modal getContainer={() => false} />
        </div>
      )
    }
    const renderComp = render(<TestComp />)
    expect(renderComp).toMatchSnapshot()
    expect(renderComp.find(classNameMap.containerBox)).toBeDefined()

    const AnotherTestComp = () => {
      const ref = React.useRef(null)
      return (
        <div ref={ref}>
          <Modal getContainer={() => false} />
        </div>
      )
    }
    const anotherRenderComp = render(<AnotherTestComp />)
    expect(anotherRenderComp).toMatchSnapshot()
    expect(anotherRenderComp.find(classNameMap.containerBox)).toBeDefined()
    expect((document.body.querySelector(classNameMap.container) as HTMLElement).style.display).toBe('')

    const ClassTestComp = () => {
      return (
        <div className={'test-get-container'}>
          <Modal getContainer={() => false} />
        </div>
      )
    }
    const classRenderComp = render(<ClassTestComp />)
    expect(classRenderComp).toMatchSnapshot()
    expect(classRenderComp.find(classNameMap.containerBox)).toBeDefined()

    const ErrorTestComp = () => {
      return (
        <div className={'test-get-container'}>
          <Modal getContainer={() => ':::'} />
        </div>
      )
    }
    const errorRenderComp = mount(<ErrorTestComp />)
    expect(errorRenderComp).toMatchSnapshot()
    expect(errorRenderComp.find(classNameMap.containerBox)).toBeDefined()
    expect(mockWarn.mock.calls[0][0]).toMatch(
      'Warning: [kdesign]-modal: getContainer"s returnValue/value is not a correct CSSSelector',
    )
  })

  it('should className use right', () => {
    const wrapper = mount(
      <Modal
        body={'test'}
        bodyClassName={'test-body'}
        bodyStyle={{ overflow: 'hidden' }}
        className={'test-over'}
        maskStyle={{ color: 'red' }}
      />,
    )
    expect(wrapper.exists('.test-body')).toBe(true)
    expect((wrapper.find('.test-body').getDOMNode() as HTMLElement).style.overflow).toBe('hidden')
    expect(wrapper.exists('.test-over')).toBe(true)
    expect((wrapper.find(classNameMap.mask).getDOMNode() as HTMLElement).style.color).toBe('red')
    unmountHelp(wrapper)
  })

  it('should width & height use right', () => {
    const wrapper = mount(<Modal height={1000} width={500} />)
    expect((wrapper.find(classNameMap.containerBox).getDOMNode() as HTMLElement).style.height).toBe('1000px')
    expect((wrapper.find(classNameMap.containerBox).getDOMNode() as HTMLElement).style.width).toBe('500px')
    unmountHelp(wrapper)
  })

  it('should destroyOnClose use right', () => {
    const wrapper = mount(<Modal destroyOnClose body={'body'} />)
    expect((wrapper.find(classNameMap.body).getDOMNode() as HTMLElement).textContent).toBe('body')
    wrapper.find(classNameMap.cancel).at(0).simulate('click')
    wrapper.update()
    expect(wrapper.exists(classNameMap.body)).toBe(false)
    unmountHelp(wrapper)
  })

  it('should footerBtnOrder use right', () => {
    const wrapper = mount(<Modal footerBtnOrder="reverse" />)
    expect((wrapper.find(classNameMap.footer).getDOMNode() as HTMLElement).style.flexDirection).toBe('row-reverse')
    unmountHelp(wrapper)
  })

  it('should config use config provider', () => {
    const modalConfig = {
      compDefaultProps: {
        Modal: {
          type: 'warning',
        },
      },
    }
    const wrapper = mount(
      <ConfigProvider value={modalConfig}>
        <Modal></Modal>
      </ConfigProvider>,
    )
    expect(wrapper.find('.kd-modal-footer')).toHaveClassName('.kd-modal-warning-footer')
  })
})
