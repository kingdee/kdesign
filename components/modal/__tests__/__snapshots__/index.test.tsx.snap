// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders correctly 1`] = `
<div
  class="kd-modal kd-modal-container kd-modal-container-show"
>
  <div
    class="kd-modal-mask"
  />
  <div
    class="kd-modal-container-box kd-modal-showline react-draggable"
    style="width:460px;height:340px;transform:translate(0px,0px)"
    tabindex="-1"
  >
    <div
      class="kd-modal-header"
    >
      <div
        class="kd-modal-title-container"
      >
        <div
          class="kd-modal-title"
        />
      </div>
      <div
        class="kd-modal-close-icon"
      >
        <i
          class="kdicon kdicon-close"
        />
      </div>
    </div>
    <div
      class="kd-modal-body"
    />
    <div
      class="kd-modal-footer kd-modal-normal-footer"
      style="flex-direction:row"
    >
      <div
        class="kd-modal-cancel-node"
      >
        <button
          class="kd-btn kd-modal-cancel-btn kd-modal-margin-btn kd-btn-second kd-btn-size-middle"
          click-animating-wave="false"
          type="button"
        >
          <span>
            取消
          </span>
        </button>
      </div>
      <div
        class="kd-modal-ok-node"
      >
        <button
          class="kd-btn kd-modal-ok-btn kd-btn-primary kd-btn-size-middle"
          click-animating-wave="false"
          type="button"
        >
          <span>
            确定
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Modal renders correctly 2`] = `
<div
  class="kd-modal kd-modal-container kd-modal-container-show"
>
  <div
    class="kd-modal-mask"
  />
  <div
    class="kd-modal-container-box kd-modal-showline react-draggable"
    style="width:460px;height:340px;transform:translate(0px,0px)"
    tabindex="-1"
  >
    <div
      class="kd-modal-header"
    >
      <div
        class="kd-modal-title-container"
      >
        <div
          class="kd-modal-title-icon modal-confirm-title-icon-bgcolor"
        />
        <div
          class="kd-modal-title"
        />
      </div>
      <div
        class="kd-modal-close-icon"
      >
        <i
          class="kdicon kdicon-close"
        />
      </div>
    </div>
    <div
      class="kd-modal-body"
    />
    <div
      class="kd-modal-footer kd-modal-confirm-footer"
      style="flex-direction:row"
    >
      <div
        class="kd-modal-cancel-node"
      >
        <button
          class="kd-btn kd-modal-cancel-btn kd-modal-margin-btn kd-btn-second kd-btn-size-middle"
          click-animating-wave="false"
          type="button"
        >
          <span>
            取消
          </span>
        </button>
      </div>
      <div
        class="kd-modal-ok-node"
      >
        <button
          class="kd-btn kd-modal-ok-btn kd-btn-primary kd-btn-size-middle"
          click-animating-wave="false"
          type="button"
        >
          <span>
            确定
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Modal renders correctly 3`] = `
<div
  class="kd-modal kd-modal-container kd-modal-container-show"
>
  <div
    class="kd-modal-mask"
  />
  <div
    class="kd-modal-container-box kd-modal-showline react-draggable"
    style="width:460px;height:340px;transform:translate(0px,0px)"
    tabindex="-1"
  >
    <div
      class="kd-modal-header"
    >
      <div
        class="kd-modal-title-container"
      >
        <div
          class="kd-modal-title-icon modal-warning-title-icon-bgcolor"
        />
        <div
          class="kd-modal-title"
        />
      </div>
      <div
        class="kd-modal-close-icon"
      >
        <i
          class="kdicon kdicon-close"
        />
      </div>
    </div>
    <div
      class="kd-modal-body"
    />
    <div
      class="kd-modal-footer kd-modal-warning-footer"
      style="flex-direction:row"
    >
      <div
        class="kd-modal-iknow-node"
      >
        <button
          class="kd-btn kd-modal-ok-btn kd-btn-second kd-btn-size-middle"
          click-animating-wave="false"
          type="button"
        >
          <span>
            我知道了
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Modal renders correctly 4`] = `
<div
  class="kd-modal kd-modal-container kd-modal-container-show"
>
  <div
    class="kd-modal-mask"
  />
  <div
    class="kd-modal-container-box kd-modal-showline react-draggable"
    style="width:460px;height:340px;transform:translate(0px,0px)"
    tabindex="-1"
  >
    <div
      class="kd-modal-header"
    >
      <div
        class="kd-modal-title-container"
      >
        <div
          class="kd-modal-title-icon modal-error-title-icon-bgcolor"
        />
        <div
          class="kd-modal-title"
        />
      </div>
      <div
        class="kd-modal-close-icon"
      >
        <i
          class="kdicon kdicon-close"
        />
      </div>
    </div>
    <div
      class="kd-modal-body"
    />
    <div
      class="kd-modal-footer kd-modal-error-footer"
      style="flex-direction:row"
    >
      <div
        class="kd-modal-iknow-node"
      >
        <button
          class="kd-btn kd-modal-ok-btn kd-btn-second kd-btn-size-middle"
          click-animating-wave="false"
          type="button"
        >
          <span>
            我知道了
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Modal renders correctly 5`] = `
<div
  class="kd-modal kd-modal-container kd-modal-container-show"
>
  <div
    class="kd-modal-mask"
  />
  <div
    class="kd-modal-container-box kd-modal-showline react-draggable"
    style="width:460px;height:340px;transform:translate(0px,0px)"
    tabindex="-1"
  >
    <div
      class="kd-modal-header"
    >
      <div
        class="kd-modal-title-container"
      >
        <div
          class="kd-modal-title"
        />
      </div>
      <div
        class="kd-modal-close-icon"
      >
        <i
          class="kdicon kdicon-close"
        />
      </div>
    </div>
    <div
      class="kd-modal-body"
    />
    <div
      class="kd-modal-footer kd-modal-normal-footer"
      style="flex-direction:row"
    >
      <div
        class="kd-modal-cancel-node"
      >
        <button
          class="kd-btn kd-modal-cancel-btn kd-modal-margin-btn kd-btn-second kd-btn-size-middle"
          click-animating-wave="false"
          type="button"
        >
          <span>
            取消
          </span>
        </button>
      </div>
      <div
        class="kd-modal-ok-node"
      >
        <button
          class="kd-btn kd-modal-ok-btn kd-btn-primary kd-btn-size-middle"
          click-animating-wave="false"
          type="button"
        >
          <span>
            确定
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Modal should draggable use right 1`] = `
<div
  class="kd-modal kd-modal-container kd-modal-container-show"
>
  <div
    class="kd-modal-mask"
  />
  <div
    class="kd-modal-container-box kd-modal-showline react-draggable"
    style="width:460px;height:340px;transform:translate(0px,0px)"
    tabindex="-1"
  >
    <div
      class="kd-modal-header"
    >
      <div
        class="kd-modal-title-container"
      >
        <div
          class="kd-modal-title"
        />
      </div>
      <div
        class="kd-modal-close-icon"
      >
        <i
          class="kdicon kdicon-close"
        />
      </div>
    </div>
    <div
      class="kd-modal-body"
    />
    <div
      class="kd-modal-footer kd-modal-normal-footer"
      style="flex-direction:row"
    >
      <div
        class="kd-modal-cancel-node"
      >
        <button
          class="kd-btn kd-modal-cancel-btn kd-modal-margin-btn kd-btn-second kd-btn-size-middle"
          click-animating-wave="false"
          type="button"
        >
          <span>
            取消
          </span>
        </button>
      </div>
      <div
        class="kd-modal-ok-node"
      >
        <button
          class="kd-btn kd-modal-ok-btn kd-btn-primary kd-btn-size-middle"
          click-animating-wave="false"
          type="button"
        >
          <span>
            确定
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Modal should draggable use right 2`] = `
<div
  class="kd-modal kd-modal-container kd-modal-container-show"
>
  <div
    class="kd-modal-mask"
  />
  <div
    class="kd-modal-container-box kd-modal-showline react-draggable"
    style="width:460px;height:340px;transform:translate(0px,0px)"
    tabindex="-1"
  >
    <div
      class="kd-modal-header"
    >
      <div
        class="kd-modal-title-container"
      >
        <div
          class="kd-modal-title"
        />
      </div>
      <div
        class="kd-modal-close-icon"
      >
        <i
          class="kdicon kdicon-close"
        />
      </div>
    </div>
    <div
      class="kd-modal-body"
    />
    <div
      class="kd-modal-footer kd-modal-normal-footer"
      style="flex-direction:row"
    >
      <div
        class="kd-modal-cancel-node"
      >
        <button
          class="kd-btn kd-modal-cancel-btn kd-modal-margin-btn kd-btn-second kd-btn-size-middle"
          click-animating-wave="false"
          type="button"
        >
          <span>
            取消
          </span>
        </button>
      </div>
      <div
        class="kd-modal-ok-node"
      >
        <button
          class="kd-btn kd-modal-ok-btn kd-btn-primary kd-btn-size-middle"
          click-animating-wave="false"
          type="button"
        >
          <span>
            确定
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Modal should getContainer use right 1`] = `
<div>
  <div
    class="kd-modal kd-modal-container kd-modal-container-show"
  >
    <div
      class="kd-modal-mask"
    />
    <div
      class="kd-modal-container-box kd-modal-showline react-draggable"
      style="width:460px;height:340px;transform:translate(0px,0px)"
      tabindex="-1"
    >
      <div
        class="kd-modal-header"
      >
        <div
          class="kd-modal-title-container"
        >
          <div
            class="kd-modal-title"
          />
        </div>
        <div
          class="kd-modal-close-icon"
        >
          <i
            class="kdicon kdicon-close"
          />
        </div>
      </div>
      <div
        class="kd-modal-body"
      />
      <div
        class="kd-modal-footer kd-modal-normal-footer"
        style="flex-direction:row"
      >
        <div
          class="kd-modal-cancel-node"
        >
          <button
            class="kd-btn kd-modal-cancel-btn kd-modal-margin-btn kd-btn-second kd-btn-size-middle"
            click-animating-wave="false"
            type="button"
          >
            <span>
              取消
            </span>
          </button>
        </div>
        <div
          class="kd-modal-ok-node"
        >
          <button
            class="kd-btn kd-modal-ok-btn kd-btn-primary kd-btn-size-middle"
            click-animating-wave="false"
            type="button"
          >
            <span>
              确定
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Modal should getContainer use right 2`] = `
<div>
  <div
    class="kd-modal kd-modal-container kd-modal-container-show"
  >
    <div
      class="kd-modal-mask"
    />
    <div
      class="kd-modal-container-box kd-modal-showline react-draggable"
      style="width:460px;height:340px;transform:translate(0px,0px)"
      tabindex="-1"
    >
      <div
        class="kd-modal-header"
      >
        <div
          class="kd-modal-title-container"
        >
          <div
            class="kd-modal-title"
          />
        </div>
        <div
          class="kd-modal-close-icon"
        >
          <i
            class="kdicon kdicon-close"
          />
        </div>
      </div>
      <div
        class="kd-modal-body"
      />
      <div
        class="kd-modal-footer kd-modal-normal-footer"
        style="flex-direction:row"
      >
        <div
          class="kd-modal-cancel-node"
        >
          <button
            class="kd-btn kd-modal-cancel-btn kd-modal-margin-btn kd-btn-second kd-btn-size-middle"
            click-animating-wave="false"
            type="button"
          >
            <span>
              取消
            </span>
          </button>
        </div>
        <div
          class="kd-modal-ok-node"
        >
          <button
            class="kd-btn kd-modal-ok-btn kd-btn-primary kd-btn-size-middle"
            click-animating-wave="false"
            type="button"
          >
            <span>
              确定
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Modal should getContainer use right 3`] = `
<div
  class="test-get-container"
>
  <div
    class="kd-modal kd-modal-container kd-modal-container-show"
  >
    <div
      class="kd-modal-mask"
    />
    <div
      class="kd-modal-container-box kd-modal-showline react-draggable"
      style="width:460px;height:340px;transform:translate(0px,0px)"
      tabindex="-1"
    >
      <div
        class="kd-modal-header"
      >
        <div
          class="kd-modal-title-container"
        >
          <div
            class="kd-modal-title"
          />
        </div>
        <div
          class="kd-modal-close-icon"
        >
          <i
            class="kdicon kdicon-close"
          />
        </div>
      </div>
      <div
        class="kd-modal-body"
      />
      <div
        class="kd-modal-footer kd-modal-normal-footer"
        style="flex-direction:row"
      >
        <div
          class="kd-modal-cancel-node"
        >
          <button
            class="kd-btn kd-modal-cancel-btn kd-modal-margin-btn kd-btn-second kd-btn-size-middle"
            click-animating-wave="false"
            type="button"
          >
            <span>
              取消
            </span>
          </button>
        </div>
        <div
          class="kd-modal-ok-node"
        >
          <button
            class="kd-btn kd-modal-ok-btn kd-btn-primary kd-btn-size-middle"
            click-animating-wave="false"
            type="button"
          >
            <span>
              确定
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Modal should getContainer use right 4`] = `
<ErrorTestComp>
  <div
    className="test-get-container"
  >
    <Modal
      getContainer={[Function]}
    >
      <Portal
        containerInfo={
          <body>
            <div
              class="kd-modal kd-modal-container kd-modal-container-hidden"
            >
              <div
                class="kd-modal-mask"
              />
              <div
                class="kd-modal-container-box kd-modal-has-container-box kd-modal-showline react-draggable"
                style="width: 460px; height: 340px; transform: translate(-230px,-170px);"
                tabindex="-1"
              >
                <div
                  class="kd-modal-header"
                >
                  <div
                    class="kd-modal-title-container"
                  >
                    <div
                      class="kd-modal-title"
                    />
                  </div>
                  <div
                    class="kd-modal-close-icon"
                  >
                    <i
                      class="kdicon kdicon-close"
                    />
                  </div>
                </div>
                <div
                  class="kd-modal-body"
                />
                <div
                  class="kd-modal-footer kd-modal-normal-footer"
                  style="flex-direction: row;"
                >
                  <div
                    class="kd-modal-cancel-node"
                  >
                    <button
                      class="kd-btn kd-modal-cancel-btn kd-modal-margin-btn kd-btn-second kd-btn-size-middle"
                      click-animating-wave="false"
                      type="button"
                    >
                      <span>
                        取消
                      </span>
                    </button>
                  </div>
                  <div
                    class="kd-modal-ok-node"
                  >
                    <button
                      class="kd-btn kd-modal-ok-btn kd-btn-primary kd-btn-size-middle"
                      click-animating-wave="false"
                      type="button"
                    >
                      <span>
                        确定
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="kd-modal kd-modal-container kd-modal-container-show"
            >
              <div
                class="kd-modal-mask"
              />
              <div
                class="kd-modal-container-box kd-modal-has-container-box kd-modal-showline react-draggable"
                style="width: 460px; height: 340px; transform: translate(-230px,-170px);"
                tabindex="-1"
              >
                <div
                  class="kd-modal-header"
                >
                  <div
                    class="kd-modal-title-container"
                  >
                    <div
                      class="kd-modal-title"
                    />
                  </div>
                  <div
                    class="kd-modal-close-icon"
                  >
                    <i
                      class="kdicon kdicon-close"
                    />
                  </div>
                </div>
                <div
                  class="kd-modal-body"
                />
                <div
                  class="kd-modal-footer kd-modal-normal-footer"
                  style="flex-direction: row;"
                >
                  <div
                    class="kd-modal-cancel-node"
                  >
                    <button
                      class="kd-btn kd-modal-cancel-btn kd-modal-margin-btn kd-btn-second kd-btn-size-middle"
                      click-animating-wave="false"
                      type="button"
                    >
                      <span>
                        取消
                      </span>
                    </button>
                  </div>
                  <div
                    class="kd-modal-ok-node"
                  >
                    <button
                      class="kd-btn kd-modal-ok-btn kd-btn-primary kd-btn-size-middle"
                      click-animating-wave="false"
                      type="button"
                    >
                      <span>
                        确定
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </body>
        }
      >
        <div
          className="kd-modal kd-modal-container kd-modal-container-show"
        >
          <div
            className="kd-modal-mask"
            onClick={[Function]}
          />
          <Draggable
            allowAnyClick={false}
            axis="both"
            bounds={false}
            cancel=".kd-modal-title-container, .kd-modal-close-icon"
            defaultClassName="react-draggable"
            defaultClassNameDragged="react-draggable-dragged"
            defaultClassNameDragging="react-draggable-dragging"
            defaultPosition={
              Object {
                "x": 0,
                "y": 0,
              }
            }
            disabled={false}
            enableUserSelectHack={true}
            handle=".kd-modal-header"
            onDrag={[Function]}
            onMouseDown={[Function]}
            onStart={[Function]}
            onStop={[Function]}
            position={
              Object {
                "x": -230,
                "y": -170,
              }
            }
            scale={1}
          >
            <DraggableCore
              allowAnyClick={false}
              cancel=".kd-modal-title-container, .kd-modal-close-icon"
              disabled={false}
              enableUserSelectHack={true}
              handle=".kd-modal-header"
              onDrag={[Function]}
              onMouseDown={[Function]}
              onStart={[Function]}
              onStop={[Function]}
              scale={1}
            >
              <div
                className="kd-modal-container-box kd-modal-has-container-box kd-modal-showline react-draggable"
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                style={
                  Object {
                    "height": 340,
                    "transform": "translate(-230px,-170px)",
                    "width": 460,
                  }
                }
                tabIndex={-1}
                transform={null}
              >
                <div
                  className="kd-modal-header"
                >
                  <div
                    className="kd-modal-title-container"
                  >
                    <div
                      className="kd-modal-title"
                    />
                  </div>
                  <div
                    className="kd-modal-close-icon"
                    onClick={[Function]}
                  >
                    <Icon
                      type="close"
                    >
                      <i
                        className="kdicon kdicon-close"
                        onClick={[Function]}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                        style={Object {}}
                      />
                    </Icon>
                  </div>
                </div>
                <div
                  className="kd-modal-body"
                />
                <div
                  className="kd-modal-footer kd-modal-normal-footer"
                  style={
                    Object {
                      "flexDirection": "row",
                    }
                  }
                >
                  <div
                    className="kd-modal-cancel-node"
                  >
                    <Button
                      className="kd-modal-cancel-btn kd-modal-margin-btn"
                      onClick={[Function]}
                      type="second"
                    >
                      <button
                        className="kd-btn kd-modal-cancel-btn kd-modal-margin-btn kd-btn-second kd-btn-size-middle"
                        click-animating-wave="false"
                        onClick={[Function]}
                        type="button"
                      >
                        <span>
                          取消
                        </span>
                      </button>
                    </Button>
                  </div>
                  <div
                    className="kd-modal-ok-node"
                  >
                    <Button
                      className="kd-modal-ok-btn"
                      onClick={[Function]}
                      type="primary"
                    >
                      <button
                        className="kd-btn kd-modal-ok-btn kd-btn-primary kd-btn-size-middle"
                        click-animating-wave="false"
                        onClick={[Function]}
                        type="button"
                      >
                        <span>
                          确定
                        </span>
                      </button>
                    </Button>
                  </div>
                </div>
              </div>
            </DraggableCore>
          </Draggable>
        </div>
      </Portal>
    </Modal>
  </div>
</ErrorTestComp>
`;
